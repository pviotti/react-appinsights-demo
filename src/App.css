.logo {
  height: 8vmin;
  pointer-events: none;
  vertical-align: middle;
}

.logos {
  margin-bottom: 50px;
}

.main {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: white;
  text-align: center;
}

.bottom-link {
  margin-top: 10vmin;
  font-size: calc(10px + 0.8vmin);
}

.page-content {
  margin-bottom: 10vmin
}
.main-text {
  font-size: calc(10px + 2vmin);
}
.react-ai-link {
  color: #61dafb;
  font-size: calc(10px + 0.8vmin);
}
button {
  margin: 2px;
}

#heart {
  padding-left: 20px;
  padding-right: 22px;
  height: 6vmin;
}
