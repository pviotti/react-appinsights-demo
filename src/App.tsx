import React, { Component } from 'react';
import NavBar from './NavBar'
import logoAI from './img/logo-ai.svg';
import logoReact from './img/logo-react.svg';
import heart from './img/heart.svg';
import './App.css';

import { Route } from 'react-router-dom';
import PageA from './DemoPageA';
import PageB from './DemoPageB';
import PageC from './DemoPageC';

export default class App extends Component {
  render() {
    return (
      <div className="App">
        <NavBar />
        <div className="main">
          <div className="page-content">
            <Route path='/' component={PageA} exact={true} />
            <Route path='/demo-a' component={PageA} />
            <Route path='/demo-b' component={PageB} />
            <Route path='/demo-c' component={PageC} />
          </div>

          <div className="bottom-link main-text">
            <div className="logos">
              <img src={logoReact} className="logo" alt="React" />
              +
              <img src={logoAI} className="logo" alt="Application Insights" />
              =
              <img src={heart} id="heart" className="logo" alt="love" />
            </div>
            <a className="react-ai-link"
              href="https://github.com/Azure/react-appinsights#react-appinsights"
              target="_blank"
              rel="noopener noreferrer"
            >
              Learn about Application Insights in React
            </a>
          </div>
        </div>
      </div>
    );
  }
}
