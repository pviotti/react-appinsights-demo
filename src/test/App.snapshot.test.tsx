import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

it('renders correctly', () => {
  const tree = renderer
    .create(<Router><App /></Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});