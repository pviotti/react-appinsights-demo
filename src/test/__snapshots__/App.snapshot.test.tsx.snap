// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark fixed-top"
  >
    <button
      aria-controls="navbarsExampleDefault"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarsExampleDefault"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarsExampleDefault"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item active"
        >
          <a
            className="nav-link"
            href="/demo-a"
            onClick={[Function]}
          >
            Page A
          </a>
        </li>
        <li
          className="nav-item active"
        >
          <a
            className="nav-link"
            href="/demo-b"
            onClick={[Function]}
          >
            Page B
          </a>
        </li>
        <li
          className="nav-item active"
        >
          <a
            className="nav-link"
            href="/demo-c"
            onClick={[Function]}
          >
            Page C
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="main"
  >
    <div
      className="page-content"
    >
      <div
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseMove={[Function]}
        onScroll={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
      >
        <div>
          <div
            className="main-text"
          >
            Demo Page A
          </div>
          <button
            className="btn btn-outline-warning btn-lg"
            onClick={[Function]}
            type="button"
          >
            Track Exception
          </button>
          <button
            className="btn btn-outline-primary btn-lg"
            onClick={[Function]}
            type="button"
          >
            Track Event
          </button>
          <button
            className="btn btn-outline-info btn-lg"
            onClick={[Function]}
            type="button"
          >
            Track Trace
          </button>
          <br />
          <button
            className="btn btn-outline-light btn-lg"
            onClick={[Function]}
            type="button"
          >
            Autocollect a Request
          </button>
          <button
            className="btn btn-outline-light btn-lg"
            onClick={[Function]}
            type="button"
          >
            Autocollect an Error
          </button>
          <br />
          ...or navigate to another page to trigger component-based page tracking.
        </div>
      </div>
    </div>
    <div
      className="bottom-link main-text"
    >
      <div
        className="logos"
      >
        <img
          alt="React"
          className="logo"
          src="logo-react.svg"
        />
        +
        <img
          alt="Application Insights"
          className="logo"
          src="logo-ai.svg"
        />
        =
        <img
          alt="love"
          className="logo"
          id="heart"
          src="heart.svg"
        />
      </div>
      <span
        className="bottom-text"
      >
        This is a demo application of Application Insights and React, using 
      </span>
      <a
        className="react-ai-link"
        href="https://github.com/Azure/react-appinsights#react-appinsights"
        rel="noopener noreferrer"
        target="_blank"
      >
        react-appinsights
      </a>
    </div>
  </div>
</div>
`;
